repl/src/main/scala/com/cloudera/livy/repl/Session.scala

```diff
@@ -210,4 +193,18 @@ class Session(interpreter: Interpreter, stateChangedCallback: SessionState => Un
 
     compact(render(resultInJson))
   }
+
+  private def executeWithJobGroup(executionCount: Int, code: String): String = {
+    Kind(interpreter.kind) match {
+      case Spark() | PySpark() | PySpark3() =>
+        val cmd = """sc.setJobGroup(%d, "Job group for statement %d")"""
+          .format(executionCount, executionCount)
+        executeCode(executionCount, cmd)
+      case SparkR() =>
+        // SparkR doesn't support statement cancelling, so doesn't need to set job group
```

Comment:
This is correct for spark2, but for spark1 we need to add SparkContext. 
Quoted from http://spark.apache.org/docs/latest/sparkr.html
```
The sc SparkContext parameter is no longer required for these functions: setJobGroup, clearJobGroup, cancelJobGroup
```